routes/login.js

```diff
@@ -0,0 +1,60 @@
+const http = require("http");
+const express = require("express")
+const bodyParser = require("body-parser")
+const bcrypt = require('bcrypt');
+const cookieParser = require("cookie-parser")
+
+const { Schema,
+    userSchema,
+    User,
+    connect,
+    sessionSchema,
+    Session,
+    postSchema,
+    Post } = require("../database.js");
+
+const router = express.Router();
+
+router.get("/", async (req, res) => {
+    try {
+        //check if there's a session. If not, render the loging main page.
+        if (!req.session) {
+            res.render('login');
+            return;
+        }
```

Comment:
Nitpick: Usually, what you do is get the "not good" paths out of the way with ifs and early returns, then do what the "happy path" is after you got all the conditions out of the way.

In this case, you would check if there's a session/user (which is unusual for the /login route) and redirect+return, then render in the last branch without ifs.

Reply:
yeah, this is what I meant, habbit made the ! mark :(